portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,79 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> comment = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jennifer!</h1>");
+    //int maxComments = Integer.parseInt(request.getParameter("max-comments"));
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    int count = 0;
+
+    ArrayList<Object> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String username = (String) entity.getProperty("username");
+      String title = (String) entity.getProperty("title");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String message = (String) entity.getProperty("message");
+
+      ArrayList<Object> comment= new ArrayList<Object>();
+      comment.add(timestamp);
+      comment.add(username);
+      comment.add(title);
+      comment.add(message);
+
+      comments.add(comment);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity("Comment");
+    
+    commentEntity.setProperty("username", request.getParameter("username"));
+    commentEntity.setProperty("title", request.getParameter("title"));
```

Comment:
You probably want to handle the use-case where getParameter returns [null](https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)) (i.e. if user didn't fill in the form) or if the value is somehow invalid. One approach is to have a wrapper function around request.getParameter that provides a reasonable default if the return value is null. See, for example, this helper function getParameter: https://github.com/googleinterns/step/blob/0218d07ca16ad8ead819a051f69edb563d030f86/walkthroughs/week-3-server/examples/text-processor/src/main/java/com/google/sps/servlets/TextProcessorServlet.java#L57